### DISCUSSION TITLE
WebDNN and npm?

### DISCUSSION THREAD

**Discussion by collinskatie** [jspsych/jsPsych#2734] [Q&A]
Hi, I am trying to use [WebDNN](https://mil-tokyo.github.io/webdnn/docs/api_reference/descriptor-runner/index.html) from within my jsPsych project.

This requires some kind of import (see the link above). I'm wondering if there is an "easy" way to import an external library like WebDNN? My sense is I might need to install WebDNN with "npm" (not too familiar with this yet, sorry). But I'm curious whether there is a more straightforward way to call packages/libraries like this? 

I'm more familiar with python, hence keen if there is some kind of easy "import" that can be done in jsPsych. Sorry if this already exists or is obvious, but wanted to check. Thanks!

**Comment by bjoluc** [comment]
Hi @collinskatie,

there's an `import` statement in modern JS which – just as you mentioned – can be used to import modules installed via NPM (JavaScript pip "equivalent"). However, it would require you to use jsPsych's [setup option 3](https://www.jspsych.org/7.3/tutorials/hello-world/#option-3-using-npm). According to the docs, WebDNN can also be embedded as a script (`<script src="./webdnn.js"></script>`), so I guess that would be the easiest approach in your situation (assuming that you are using jsPsych setup options 1 or 2).

Cheers!
